match/tensordata.py

```diff
@@ -0,0 +1,432 @@
+                    self.__out_of_bounds_coords(index)
+                    self._data[
+                        self.__multi_to_single_rank_translation(index)
+                    ]._item = self.dtype(value)
+            elif isinstance(value, TensorData):
+                for i, index in enumerate(itertools.product(*possible_indices)):
+                    self.__out_of_bounds_coords(index)
+                    self._data[
+                        self.__multi_to_single_rank_translation(index)
+                    ]._item = value._data[i]._item
+
+    # TODO(SRM47) Update the repr to make it look like PyTorch
+    def __repr__(self):
+        return self._data.__repr__() if self._item is None else self._item.__repr__()
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __translate(self, *shape: int) -> tuple:
+        shape = shape[0]
+        res = [0] * len(self.shape)
+        for i in range(len(res) - 1, -1, -1):
+            res[i] = (
+                0 if self.shape[i] == 1 else shape[i + len(shape) - len(self.shape)]
+            )
+        return tuple(res)
+
+    def __validate_broadcast(self, shape):
+        """Helper function to determine whether self TensorData can be broadcasted 
+        to desired shape."""
+        for s1, s2 in zip(reversed(self.shape), reversed(shape)):
+            if not (s1 == 1 or s2 == 1 or s1 == s2):
+                raise ValueError("Can't broadcast")
+
+    def broadcast(self, *shape: int):
+        """Broadcasts the TensorData to the desired shape.
+
+        Args:
+            shape: The desired shape of the tensor.
+
+        Returns:
+            A new tensor with the desired shape.
+
+        Raises:
+            ValueError: If the tensor cannot be broadcast to the desired shape.
+        """
+        self.__validate_broadcast(shape)
+
+        if self.shape == shape:
+            return self
+        
+        new_tensor = TensorData(*shape[len(self.shape) - 1 :], dtype=self.dtype)
+        possible_indices = [range(dim) for dim in new_tensor.shape]
+        for i, new_tensor_index in enumerate(itertools.product(*possible_indices)):
+            translated_index = self.__translate(new_tensor_index)
+            single_index = self.__multi_to_single_rank_translation(translated_index)
+            new_tensor._data[i]._item = self._data[single_index].item()
+
+        remianing_dimensions = shape[: len(self.shape) - 1]
+
+        new_data = []
+        for _ in range(prod(remianing_dimensions)):
+            new_data.extend(deepcopy(new_tensor._data))
+
+        new_tensor._data = new_data
+        new_tensor.__reshape(shape)
+
+        return new_tensor
+
+    def unbroadcast(self, *shape: int):
+        """Return a new TensorData unbroadcast from current shape to desired shape."""
+        ...
+
+    @staticmethod
+    def randn(*shape: int) -> 'TensorData':
+        """Helper method to quickly create a List2D object with random values."""
+        ...
```

Comment:
Why is this one static instead of being more like `ones_` and `zeros_`?

Reply:
I copied this function from the List2D.py class. Would you like me to change it?